src/hooks/useIsValidExecution.ts

```diff
@@ -4,14 +4,45 @@ import type { EthersError } from '@/utils/ethers-utils'
 
 import useAsync from './useAsync'
 import ContractErrorCodes from '@/services/contracts/ContractErrorCodes'
-import { useSafeSDK } from './coreSDK/safeCoreSDK'
+import { type SafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import { getAndValidateSafeSDK } from '@/services/tx/tx-sender/sdk'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { createWeb3, useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import EthersAdapter from '@safe-global/safe-ethers-lib'
+import { ethers } from 'ethers'
+import useWallet from '@/hooks/wallets/useWallet'
+import { type JsonRpcProvider } from '@ethersproject/providers'
+import { type ConnectedWallet } from '@/services/onboard'
 
 const isContractError = (error: EthersError) => {
   if (!error.reason) return false
 
   return Object.keys(ContractErrorCodes).includes(error.reason)
 }
 
+// Monkey patch the signerProvider to proxy requests to the "readonly" provider if on the wrong chain
+const getPatchedSignerProvider = (
+  wallet: ConnectedWallet,
+  chainId: SafeInfo['chainId'],
+  readOnlyProvider: JsonRpcProvider,
+) => {
+  const signerProvider = createWeb3(wallet.provider)
+
+  if (wallet.chainId !== chainId) {
+    const signerMethods = ['eth_accounts']
+    const originalSend = signerProvider.send
+
+    signerProvider.send = (request, ...args) => {
+      if (!signerMethods.includes(request)) {
+        return readOnlyProvider.send.call(readOnlyProvider, request, ...args)
+      }
+      return originalSend.call(signerProvider, request, ...args)
+    }
+  }
```

Comment:
The hook seems to over-request the RPC a lot now. I get around 70 requests when trying to execute. I suspect it has to do with how we overwrite the `.send` call. Any ideas @katspaugh?

Reply:
It is creating a new provider on every re-render. I'd look into memoisation and/or cleanup of the previous provider.

Reply:
I checked and the hook only runs once. In the network requests I can see that its the `.send.call` we do probably because we actually call the function instead of just assigning I assume

Reply:
As Aaron suggested, it seems you might be creating a recursion by repeatedly overwriting the send method. The first time you patch it, the original send is the actual original send, but on subsequent calls it will be the patched version.

Reply:
I just pushed a fix for #1774 by switching from using `sdk.isValidTransaction` to using the contract directly. This also fixed the over-fetching issue.